src/scripts/browserDeviceProfile.js

```diff
@@ -346,26 +380,19 @@ import browser from './browser';
         const canPlayAc3VideoAudioInHls = supportsAc3InHls(videoTestElement);
 
         // Transcoding codec is the first in hlsVideoAudioCodecs.
-        // Prefer AAC, MP3 to other codecs when audio transcoding.
-        if (canPlayAacVideoAudio) {
-            videoAudioCodecs.push('aac');
-            hlsInTsVideoAudioCodecs.push('aac');
-            hlsInFmp4VideoAudioCodecs.push('aac');
-        }
-
+        // AAC, MP3, AC-3 and EAC-3 are supported in HLS
+        // Here they are sorted based on the quality of their encoders in ffmpeg
         if (supportsMp3VideoAudio) {
             videoAudioCodecs.push('mp3');
 
-            // PS4 fails to load HLS with mp3 audio
-            if (!browser.ps4) {
+            // PS4 fails to load HLS with MP3 audio
+            // Only prefer MP3 when the client wants stereo or mono to prevent downmixing
+            if (!browser.ps4 && physicalAudioChannels <= 2) {
                 hlsInTsVideoAudioCodecs.push('mp3');
+                hlsInFmp4VideoAudioCodecs.push('mp3');
             }
-
-            hlsInFmp4VideoAudioCodecs.push('mp3');
         }
 
-        // For AC3/EAC3 remuxing.
-        // Do not use AC3 for audio transcoding unless AAC and MP3 are not supported.
```

Comment:
It'd be useful to know why you're removing comments.

Reply:
It's not only used for remuxing and AC-3 is preferred to AAC.